### PR TITLE
273 - Updated Item association

### PR DESCRIPTION
**Pull request by mohit2530**
[273 - Updated item association](https://github.com/earmuff-jam/pulse/pull/275/commits/aa19f4e046df14b1e6d8baaee4e856e811a449db)

- Added the ability to remove item association between
categories, maintenance plans and assets. Added backend support
for this effort for both categories and maintenance plans and added
ui integration to support this feature.

- Removed use case of Datatable and used tableComponent instead because
of our various customization needs, that needed to be supported by Datatable.
For now, using our own tableComponent provided better support than using
the custom solution of muiv5 data table.

- Updated tech debt and cleanup on components such as removing console.log
across the newly created components. Also updated inventories api handler
to provide a consistent updator name instead of updater_name vs updator that
we use in categories and maintenance items.

Closes https://github.com/earmuff-jam/pulse/issues/271

### CODE REVIEW

File: apilayer/handler/CategoriesHandler.go [hunk 1]
```diff
@@ -273,6 +273,45 @@ func AddItemsInCategory(rw http.ResponseWriter, r *http.Request, user string) {
...
 
+// RemoveAssociationFromCategory ...
+// swagger:route POST /api/v1/category/remove/items Categories RemoveAssociationFromCategory
+//
+// # Removes association between the category and asset item
+//
+// Parameters:
+//   - +name: CategoryItemRequest
+//     in: body
+//     description: The object containing the array of assets to update with the userID
```
Comment (earmuff-jam) [inline apilayer/handler/CategoriesHandler.go:284]: wrong description

File: apilayer/handler/MaintenancePlan.go [hunk 2]
```diff
@@ -224,6 +224,45 @@ func AddItemsInMaintenancePlan(rw http.ResponseWriter, r *http.Request, user str
...
 
+// RemoveAssociationFromMaintenancePlan ...
+// swagger:route POST /api/v1/plan/remove/items MaintenancePlans RemoveAssociationFromMaintenancePlan
+//
+// # Removes association between the maintenance plan and asset item
+//
+// Parameters:
+//   - +name: MaintenanceItemRequest
+//     in: body
+//     description: The object containing the array of assets to be removed
```
Comment (earmuff-jam) [inline apilayer/handler/MaintenancePlan.go:235]: wrong description

File: frontend/src/features/MaintenancePlanItemDetails/maintenancePlanItemSlice.js [hunk 3]
```diff
@@ -54,6 +54,21 @@ const maintenancePlanItemSlice = createSlice({
       state.loading = false;
       state.error = '';
     },
+    removeItemsFromMaintenancePlan: (state) => {
+      state.loading = true;
+      state.error = '';
+    },
+    removeItemsFromMaintenancePlanSuccess: (state, action) => {
+      const removedItems = action.payload;
+      const filteredItemsFromCategoryList = state.itemsInMaintenancePlan.filter((v) => !removedItems.includes(v.id));
```
Comment (earmuff-jam) [inline frontend/src/features/MaintenancePlanItemDetails/maintenancePlanItemSlice.js:63]: rename this to properly reflect the state

**Review by earmuff-jam — CHANGES_REQUESTED** [review]
Just some minor comments to flush out. I noticed that we don't add confirmation modal to actions when we delete the selected asset from either category or maintenance plan. Lets add that as well. Also when we delete a selected asset from `category` or `maintenance plan` and try to add it back immediately, we get a failure. We need to investigate why. Also we also need to add `<Snackbar />` components after adding / removing items. Lets fix those issues.

**Comment by mohit2530** [comment]
Added Confirmation box modal before deletion of each items.
image

**Review by earmuff-jam — APPROVED** [review]
All changes looks good.

**Pull request merged**